src/playercards/AllCardsBag.ttslua

```diff
@@ -27,20 +184,49 @@ end
 function clearIndexes()
   indexingDone = false
   cardIdIndex = { }
-  classAndLevelIndex = { }
-  classAndLevelIndex["Guardian-upgrade"] = { }
-  classAndLevelIndex["Seeker-upgrade"] = { }
-  classAndLevelIndex["Mystic-upgrade"] = { }
-  classAndLevelIndex["Survivor-upgrade"] = { }
-  classAndLevelIndex["Rogue-upgrade"] = { }
-  classAndLevelIndex["Neutral-upgrade"] = { }
-  classAndLevelIndex["Guardian-level0"] = { }
-  classAndLevelIndex["Seeker-level0"] = { }
-  classAndLevelIndex["Mystic-level0"] = { }
-  classAndLevelIndex["Survivor-level0"] = { }
-  classAndLevelIndex["Rogue-level0"] = { }
-  classAndLevelIndex["Neutral-level0"] = { }
+  allCardsIndex = { }
+  otherIndex = { }
+  classIndex = { }
+  classIndex["guardian"] = { }
+  classIndex["seeker"] = { }
+  classIndex["rogue"] = { }
+  classIndex["mystic"] = { }
+  classIndex["survivor"] = { }
+  classIndex["neutral"] = { }
+  classIndex[1] = { }
+  classIndex[2] = { }
+  classIndex[3] = { }
+  levelIndex = { }
+  levelIndex[0] = { }
+  levelIndex[1] = { }
+  levelIndex[2] = { }
+  levelIndex[3] = { }
+  levelIndex[4] = { }
+  levelIndex[5] = { }
   cycleIndex = { }
+  traitsIndex = { }
+  willpowerIconsIndex = { [0] = { } }
+  intellectIconsIndex = { [0] = { } }
+  combatIconsIndex    = { [0] = { } }
+  agilityIconsIndex   = { [0] = { } }
+  wildIconsIndex      = { [0] = { } }
+  typeIndex = { }
+  slotIndex = { }
+  slotIndex["arcane"] = { }
+  slotIndex["arcane x2"] = { }
+  slotIndex["ally"] = { }
+  slotIndex["accessory"] = { }
+  slotIndex["body"] = { }
+  slotIndex["tarot"] = { }
+  slotIndex["hand"] = { }
+  slotIndex["hand x2"] = { }
+  slotIndex[1] = { }
+  slotIndex[2] = { }
+  slotIndex[3] = { }
```

Comment:
It's bad practice to have number and string keys for the same table. This should be split into a separate table.

Reply:
I agree it's a bit messy, but since this isn't really something that's going to be touched by any other function we can hopefully leave it as is?  If this becomes a separate table, the Filter Card Search would need an entirely separate operator to track them (because I use `_G` for it in the function below).

Alternatively, if it is too intolerable, I could maybe switch to just strings of the numbers instead?

Reply:
For clarity, currently the code for the index looks like this:
```lua
local filterOptions = {
  y = "cycle",
  f = "class",
  k = "traits",
  o = "cost",
  w = "willpowerIcons",
  c = "combatIcons",
  i = "intellectIcons",
  a = "agilityIcons",
  d = "wildIcons",
  t = "type",
  p = "level",
  z = "slot",
  v = "permanent",
  u = "uses",
  q = "specialCriteria"
}
...
matchingCards = allCardsBagApi.getCardsByData(filterOptions[filter], value, includeOthers)
```


So the issue here really is just the namespace of the variables for the tables in `AllCardsBag` gets very ugly if we make this change, because they have to be consistent in order to be usable with `_G`.  So that means all the tables that use numeric indices would have to be appended with `_number` or something, even the ones that have exclusively numeric indices like the icon ones.  

Personally I don't see some mixed indices in the tables as a particularly big problem, since in practical terms the only limitation _I think_ it presents is not being able to use `ipairs`?  I guess I'm just not seeing the downside?

Reply:
Isn't this just mixing two indices in one?
`slotIndex` should be holding all cards that take up a specific slot in each subtable.
The "1, 2 and 3" subtables are technically an additional index for the slot count.
I don't see why we have to combine that into a single index.